### DISCUSSION TITLE
Any Ideas on How to do this?  Cannot figure out how to get custom object mapper settings to be used

### DISCUSSION THREAD

**Discussion by golfingal72** [Netflix/dgs-framework#249] [Q&A]
I am running a small DGS app and have a Configuration class that is configuring a custom object mapper, but no matter what I try it doesn't seem to be getting recognized??  I have a 'null' field in the output that I am trying to suppress, but no matter what i try, it still gets serialized as null..  The configuration class is below and I have debug in it showing it does get executed from the SpringBootApplication context loading.  

Help!!

`{
  "data": {
    "federatedFavoriteProduct": {
      "id": "581d20a052d9d339cbe5c30a",
      "productId": 606506412,
      "product": {
        "id": 606506412,
        "name": "TestProduct",
        "description": null
      }
    }
  }
}`

`@Configuration
@Slf4j
public class ProductDgsConfig {

    @Bean
    public Jackson2ObjectMapperBuilderCustomizer jsonCustomizer() {
        log.info("Customize Jackson Object Mapper");
        SimpleModule serializationModule = new SimpleModule("ShopStyleModule");
        serializationModule.addSerializer(new CustomDateSerializer());

        return builder -> builder.serializationInclusion(JsonInclude.Include.NON_EMPTY)
                .modules(serializationModule, new JavaTimeModule(), new KotlinModule())
                .featuresToEnable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS,
                        MapperFeature.DEFAULT_VIEW_INCLUSION)
                .featuresToDisable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS)
                .failOnEmptyBeans(false);

        // .configure(objectMapper());

    }

    /**
     * Custom Object Mapper for responses
     *
     * @return objectMapper
     */
    @Bean
    @Primary
    public ObjectMapper objectMapper(Jackson2ObjectMapperBuilder builder) {

        log.info("CREATING CUSTOM OBJECT MAPPER");
        // Module to group our custom serializers
        SimpleModule serializationModule = new SimpleModule("ShopStyleModule");
        serializationModule.addSerializer(new CustomDateSerializer());

        builder.serializationInclusion(JsonInclude.Include.NON_EMPTY)
                .modules(serializationModule, new JavaTimeModule(), new KotlinModule())
                .featuresToEnable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS,
                        MapperFeature.DEFAULT_VIEW_INCLUSION)
                .featuresToDisable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS)
                .failOnEmptyBeans(false);
        ObjectMapper newobjectMapper = builder.build();
        log.info("ObjectMapper: {}", ToStringBuilder.reflectionToString(newobjectMapper));

        return newobjectMapper;
    }

    private static class CustomDateSerializer extends StdScalarSerializer<Date>
            implements ContextualSerializer {

        private final FastDateFormat dateFormatter = FastDateFormat.getInstance("yyyy-MM-dd");

        public CustomDateSerializer() {
            super(Date.class);
        }

        @Override
        public JsonSerializer<?> createContextual(SerializerProvider prov, BeanProperty property)
                throws JsonMappingException {
            // This is the place where we can read annotations to modify the default behavior
            if (property != null) {
                JsonFormat.Value format =
                        prov.getAnnotationIntrospector().findFormat(property.getMember());
                if (format != null) {
                    TimeZone tz = format.getTimeZone();
                    String pattern = format.getPattern();
                    if (pattern.length() > 0) {
                        SimpleDateFormat df = new SimpleDateFormat(pattern);
                        if (tz != null) {
                            df.setTimeZone(tz);
                        }
                        return new DateSerializer(false, df);
                    }
                }
            }
            return this;
        }

        @Override
        public JsonNode getSchema(SerializerProvider provider, Type typeHint)
                throws JsonMappingException {
            ObjectNode objectNode = createSchemaNode("object", true);
            objectNode.replace("date", createSchemaNode("string", true));
            objectNode.replace("timestamp", createSchemaNode("number", false));
            objectNode.replace("friendly", createSchemaNode("string", true));
            return objectNode;
        }

        @Override
        public void acceptJsonFormatVisitor(JsonFormatVisitorWrapper visitor, JavaType typeHint)
                throws JsonMappingException {
            // Not implemented
            return;
        }

        @Override
        public void serialize(Date value, JsonGenerator jgen, SerializerProvider provider)
                throws IOException, JsonGenerationException {
            jgen.writeStartObject();
            jgen.writeStringField("date", dateFormatter.format(value));
            jgen.writeNumberField("timestamp", value.getTime() / 1000);
            jgen.writeStringField("friendly",
                    FriendlyDateFormat.getInstance(ConfigProperties.getLocale()).format(value));

            jgen.writeEndObject();
        }
    }
}`

**Comment by golfingal72** [comment]
@paulbakker ?? how can i fix this?

**Reply by paulbakker** [reply]
We don't currently support providing a custom mapper. The mapper is defined in the [DefaultDgsQueryExecutor](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultDgsQueryExecutor.kt#L65)

**Reply by golfingal72** [reply]
Can I write a Ticket to support a custom mapper?  I do NOT want NULLS/EMPTY fields in my responses... I hate to branch from the base codebase

**Comment by EugenMayer** [comment]
Any reason not to consider implementing something similar as spring boot MvC offers with `Jackson2ObjectMapperBuilderCustomizer` in DGS?

This would offer this very flexibility. It would also open integration points for frameworks like https://github.com/Blazebit/blaze-persistence when implementing support for Mutations - offering an integration for DGS like e.g. is offered for spring MvC with https://github.com/Blazebit/blaze-persistence/tree/master/integration/spring-data/webmvc 

This would open up for integrations into DGS, which might be valuable?

**Reply by golfingal72** [reply]
I am revisiting this ...  Is there no way to provide this feature or at least have graphQL config variables in the application.properties to customize the object mapper features...  this would be so simple to do...

**Reply by golfingal72** [reply]
payloads with NULL fields are so bloated....

**Reply by golfingal72** [reply]
@paulbakker ping

**Reply by jord1e** [reply]
You can contribute #813

**Reply by paulbakker** [reply]
Is this specifically for serializing the result back to the client?
